plugins/http_plugin/include/eosio/http_plugin/beast_http_session.hpp

```diff
@@ -164,6 +168,28 @@ class beast_http_session : public detail::abstract_conn {
       }
    }
 
+private:
+   void send_100_continue_response(bool do_continue) {
+      auto res = std::make_shared<http::response<http::empty_body>>();
+         
+      res->version(11);
+      if (do_continue) {
+         res->result(http::status::continue_);
+         continue_state_ = continue_state_t::read_body;   // after sending the continue response, just read the body with the same parser
+      } else {
+         res->result(http::status::unauthorized);
```

Comment:
you are right, even though I saw it in an example it is not the right code. see [this](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/417). Thanks!

Reply:
Actually, my bad, `http::status::unauthorized` was correct, see [this](https://www.rfc-editor.org/rfc/rfc7231#section-5.1.1), and esp:
![Screenshot from 2023-03-01 16-53-29](https://user-images.githubusercontent.com/1496907/222273476-a30be624-ec48-4ebf-b7ed-9e644e0d3c9f.png)

Reply:
I tend to think 417 is better. The 401 in the paragraph you quoted above was just an example of many possible errors. By the standard, https://www.rfc-editor.org/rfc/rfc7235#section-3.1, "The 401 (Unauthorized) status code indicates that the request has not been applied because it lacks valid authentication credentials for the target resource.". A body which is too big is not client's error. It is just this server cannot fulfill the request.

Reply:
From my reading of the spec above (first paragraph), the 417 is meant to be used when the "Expect" requested is not supported, like something different from "100-continue".

If it was meant to be the correct response when the length is too large, they would have used it in the example, instead of suggesting 401 or 405.

Reply:
It is the spec's fault, which contradicts itself. 405 is for method not allowed, but we allow the method. If we send out 405, the client would stop sending 100 expect to us, even for small bodies. I just feel 401 always involves some authentication failures. The 417 section is terse and does sound 417 should be used
https://www.rfc-editor.org/rfc/rfc7231#section-6.5.14

I think the first paragraph above is just for one case in which 417 can be sent out.

Maybe we should ask our Engineering channel for opinions.

Reply:
Also see [here](https://www.bram.us/2020/04/14/about-the-http-expect-100-continue-header/)
<img width="498" alt="image" src="https://user-images.githubusercontent.com/1496907/222471890-0479ce65-d816-4d9a-9f23-a5e678c61a50.png">

Reply:
It is in the [spec](https://www.rfc-editor.org/rfc/rfc7231#section-5.1.1) as well:

<img width="413" alt="image" src="https://user-images.githubusercontent.com/1496907/222472525-251e871a-5e18-4b0d-967b-6ca667ffb02a.png">

Reply:
OK. Keep 401. It's good we dug this and documented the rational.

Reply:
Thanks @linh2931 for the review and approval!